### PR TITLE
app-text/cherrytree: Bump version to 0.99.53, Drop old versions 0.99.37 0.99.47-r1 0.99.48 0.99.52

### PR DESCRIPTION
**Pull request by flint2**
Signed-off-by: Guillermo Joandet <gjoandet@gmail.com>

### CODE REVIEW

**Comment by scardracs** [comment]
Why didn't you bump and drop it on the same PR?

**Comment by flint2** [comment]
> Why didn't you bump and drop it on the same PR?

I thought it will be easier for maintainers, now is all in one PR

File: app-text/cherrytree/cherrytree-0.99.52.ebuild [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	dev-libs/libfmt:=
+	dev-libs/fribidi
+	dev-libs/spdlog
+	net-misc/curl
+	x11-libs/cairo
+	x11-libs/gtk+:3
+	x11-libs/pango
+	x11-libs/vte:2.91"
+DEPEND="${PYTHON_DEPS}
+${RDEPEND}
```
Comment (juippis) [inline app-text/cherrytree/cherrytree-0.99.52.ebuild:53]:
```suggestion
	${RDEPEND}
```

File: app-text/cherrytree/cherrytree-0.99.52.ebuild [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+"
+BDEPEND="
+	virtual/pkgconfig
+	nls? (
+		sys-devel/gettext
+	)
```
Comment (juippis) [inline app-text/cherrytree/cherrytree-0.99.52.ebuild:59]:
```suggestion
	nls? ( sys-devel/gettext )
```

File: app-text/cherrytree/cherrytree-0.99.52.ebuild [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	# disable compress man pages
+	sed -i -e \
+		'/install(FILES/s|${MANFILE_FULL_GZ}|${CMAKE_SOURCE_DIR}/data/cherrytree.1|' \
+		CMakeLists.txt || die
+
+	if [[ ${PV} != *9999 ]]; then
+		sed -i \
+			-e "/^set(CT_VERSION/s|\"\(.*\)\"|\"${PV}\"|" \
+			CMakeLists.txt || die
+	fi
```
Comment (juippis) [inline app-text/cherrytree/cherrytree-0.99.52.ebuild:72]: It doesn't look like this is doing anything, could you check if this is still relevant?
```
 * SED: the following did not cause any changes
 *         sed -i -e "/^set(CT_VERSION/s|\"\(.*\)\"|\"${PV}\"|" CMakeLists.txt || die;
 * no-op: -e /^set(CT_VERSION/s|"\(.*\)"|"0.99.52"|
```

File: app-text/cherrytree/cherrytree-0.99.52.ebuild [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+	dev-db/sqlite:3
+	dev-libs/glib:2
+	dev-libs/libfmt:=
+	dev-libs/fribidi
+	dev-libs/spdlog
+	net-misc/curl
+	x11-libs/cairo
+	x11-libs/gtk+:3
+	x11-libs/pango
+	x11-libs/vte:2.91"
```
Comment (juippis) [inline app-text/cherrytree/cherrytree-0.99.52.ebuild:51]: Looks like you're missing some deps:
```
 * VDB: detected possibly incorrect RDEPEND (app-text/cherrytree-0.99.52)
 *                 > dev-cpp/atkmm
 *                 > dev-cpp/cairomm
 *                 > dev-libs/libsigc++:2
 *                 > dev-libs/libxml2
```

Comment (juippis) [inline app-text/cherrytree/cherrytree-0.99.52.ebuild:51]: Are you sure exactly this vte is necessary?

**Comment by juippis** [comment]
> > Why didn't you bump and drop it on the same PR?
> 
> I thought it will be easier for maintainers, now is all in one PR

Most likely the Manifest files will conflict with each other when one gets merged before the other, so all related commits in a single PR is therefore better. It's indeed easier to merge a removing PR faster since you don't need to build-test it, but as said, then the version bump won't apply to git-tree anymore cleanly causing a bit more manual work which is annoying...

File: app-text/cherrytree/cherrytree-0.99.52.ebuild (hunk 4 above)
Comment (flint2) [reply app-text/cherrytree/cherrytree-0.99.52.ebuild:51]: Will be changed to gui-libs/vte on next commit

File: app-text/cherrytree/Manifest [hunk 5]
```diff
@@ -1,4 +1,4 @@
-DIST cherrytree-0.99.37.tar.gz 17983743 BLAKE2B 42e5464f4cba4364dbbdabe5daaf80517a62f3f4b2c1c3efa764e9b10dfae492bf335403d24a7a33d9b7c8cd55a0c55ad359ffca1f639fa7622e00a416defd60 SHA512 2413924d03c65dfad820bc2b1e345b7dd74b69d11a10ee0d333400936eb31368f53f2aee77a3dd708adb3d4b91578c0499766bca4c85537822e9c4b3d41c6800
 DIST cherrytree_0.99.47.tar.xz 2457904 BLAKE2B 1072de8ed62757a873a258dc9ab1e9e186128b520e7f02e0a2cd8e7e0173232af7db85e3293efd21b577e51b6f5e55cca0cf35631e34279e32dc07836335b742 SHA512 2611be4c24a188793aa6785d96b33ec91c0957b30616352e1cb2cd12f296e39c29666d37a03016be6c934898bf6deb98af74c2c345d9c70f138f62ac6b10f023
 DIST cherrytree_0.99.48.tar.xz 3474468 BLAKE2B b60cc2ffa4f5bd6e91e2a8b34811ad6977d46bdd02c7c1b2bfcdbf4df6107753a2390bc6a15daae8d5268af017ff52d217ab79ed5b4d5d107ef736257999b865 SHA512 eb4690bbc5865b9a0c7d0cb11341fe5ac17f11d417ad765b29784083a83f18c06a39b5b738838e3d57754de25adf1861ea4a6ad472e01a895e8fce0af0964926
```
Comment (scardracs) [inline app-text/cherrytree/Manifest:2]: Drop all 3 versions too, if not required by specific packages, and ask for a stablereq of 0.99.51 when this PR will be merged. Having lots of versions of the same package is not always a good idea (except for some packages like mariadb or database manager in general)

File: app-text/cherrytree/cherrytree-0.99.52.ebuild [hunk 6]
```diff
@@ -1,23 +1,25 @@
...
 	S="${WORKDIR}/${P}/future"
 else
-	SRC_URI="https://github.com/giuspen/cherrytree/archive/refs/tags/${PV}.tar.gz -> ${P}.tar.gz"
+	SRC_URI="https://github.com/giuspen/cherrytree/releases/download/${PV}/${P/-/_}.tar.xz"
+	S="${WORKDIR}"/${P/-/_}
+
 	KEYWORDS="~amd64 ~x86"
-# 	S="${WORKDIR}/${PN}-${COMMIT}/future"
+	# S="${WORKDIR}/${PN}-${COMMIT}/future"
 fi
```
Comment (scardracs) [inline app-text/cherrytree/cherrytree-0.99.52.ebuild:23]: That package does not have any live-ebuild so you can proceed in 2 ways: drop that if...else...fi or create (and maintain) the live-ebuild

File: app-text/cherrytree/cherrytree-0.99.52.ebuild [hunk 7]
```diff
@@ -30,8 +32,7 @@ IUSE="nls test"
 # Has deps that aren't available in ::gentoo repo
 RESTRICT="test"
 
-RDEPEND="
-	app-i18n/uchardet
+RDEPEND="app-i18n/uchardet
```
Comment (scardracs) [inline app-text/cherrytree/cherrytree-0.99.52.ebuild:35]: Why did you change it?

File: app-text/cherrytree/cherrytree-0.99.52.ebuild [hunk 8]
```diff
@@ -40,29 +41,24 @@ RDEPEND="
...
+	x11-libs/vte:2.91"
+DEPEND="${PYTHON_DEPS}
+${RDEPEND}
+"
+BDEPEND="
 	virtual/pkgconfig
 	nls? (
-		dev-util/intltool
 		sys-devel/gettext
 	)
```
Comment (scardracs) [inline app-text/cherrytree/cherrytree-0.99.52.ebuild:59]: `nls? ( sys-devel/gettext )`

File: app-text/cherrytree/cherrytree-0.99.52.ebuild (hunk 6 above)
Comment (ajakk) [reply app-text/cherrytree/cherrytree-0.99.52.ebuild:23]: I'm not sure there's anything wrong with keeping dead live-specific code in the ebuild as it provides an easy point from which to make a live ebuild if desired in the future. No reason to require that being done now.

File: app-text/cherrytree/cherrytree-0.99.52.ebuild (hunk 7 above)
Comment (ajakk) [reply app-text/cherrytree/cherrytree-0.99.52.ebuild:35]: To match the rest of the dependency declarations?

File: app-text/cherrytree/cherrytree-0.99.52.ebuild (hunk 3 above)
Comment (flint2) [reply app-text/cherrytree/cherrytree-0.99.52.ebuild:72]: references to dev version 9999 will be removed on next commit

File: app-text/cherrytree/cherrytree-0.99.52.ebuild (hunk 4 above)
Comment (flint2) [reply app-text/cherrytree/cherrytree-0.99.52.ebuild]: dev-libs/libsigc++:2, dev-cpp/atkmm and dev-cpp/cairomm are required by dev-cpp/gtkmm

**Comment by thesamesam** [comment]
Thanks. I split the "drop old" into a new commit.
